server/src/services/address.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Transaction } from 'sequelize/types';
+import User from '../models/user.model';
+import UserAddress from '../models/userAddress.model';
+
+import { ICreateAddress, IUpdateAddress } from '../interfaces/addressService.interface';
+
+class AddressService {
+  private async _updateUserDefaultAddressByUserId(
+    transaction: Transaction,
+    user_id: string,
+    address_id: number | null
+  ) {
+    const [userUpdateSuccess] = await User.update(
+      { default_address: address_id },
+      {
+        where: {
+          id: user_id,
+        },
+        transaction,
+      }
+    );
+    if (!userUpdateSuccess) {
+      throw Error(); // unexpected error
+    }
+  }
+
+  public async createAddress(transaction: Transaction, user_id: string, address: ICreateAddress): Promise<UserAddress> {
+    const isDefaultAddress: boolean | undefined = address.default_address;
+    const newUserAddress: UserAddress = (await UserAddress.create({ ...address, user_id }, { transaction })).get();
```

Comment:
```suggestion
    const newUserAddress: UserAddress = await UserAddress.create({ ...address, user_id }, { transaction });
```
get을 안써도 이미 UserAddress로 반환되진 않나요?

Reply:
sequelize 객체가 아닌 일반 자바스크립트 객체를 얻어오기 위해 사용했던것인데 raw와 plain을 빼면서 sequelize객체여도 반환하는데 문제가 없다는걸 깨달았습니다!
저 get 부분은 제거 해 놓겠습니다!